icu4j/main/classes/core/src/com/ibm/icu/number/NumberFormatterImpl.java

```diff
@@ -192,6 +182,8 @@ private static MicroPropsGenerator macrosToMicroGenerator(MacroProps macros, Mic
         }
         boolean isCldrUnit = !isCurrency && !isBaseUnit &&
             (unitWidth == UnitWidth.FULL_NAME || !(isPercent || isPermille));
+        boolean isMixedUnit = isCldrUnit && macros.unit.getType() == null &&
+                macros.unit.getComplexity() == MeasureUnit.Complexity.MIXED;
```

Comment:
You need the logic in this file that sets the rounder to bogus/pass-through if `isMixedUnit` is true.  There may also be some other pieces missing in this file.  Please check the C++ diff, and make sure that this code path is properly tested.  @hugovdm

Reply:
Please add this as a TODO.

Reply:
done

Reply:
This is icu-units#95 - TODO(icu-units#95) would help cross-reference.

Reply:
Added the issue number in 55c783d3b5